packages/calcite-components/src/controllers/usePreventDocumentScroll.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * A controller to manage preventing document scroll when a component is opened.
+ * This utility ensures that the document's scroll behavior is disabled when a component
+ * with `preventDocumentScroll` set to `true` is opened, and restores the scroll behavior
+ * when the component is closed or disconnected.
+ *
+ * @module usePreventDocumentScroll
+ */
+
+import { makeGenericController } from "@arcgis/components-controllers";
+import { LitElement } from "@arcgis/lumina";
+
+let count: number = 0;
+let initialDocumentOverflowStyle: string = "";
+
+/**
+ * Interface representing a component that can prevent document scrolling.
+ */
+interface PreventDocumentScrollComponent extends LitElement {
+  /**
+   * Indicates whether the component is currently opened.
+   */
+  opened: boolean;
+
+  /**
+   * Indicates whether the component should prevent document scrolling when opened.
+   */
+  preventDocumentScroll: boolean;
+}
+
+/**
+ * Creates a controller to manage document scroll prevention for a component.
+ *
+ * This controller tracks the number of components requesting scroll prevention
+ * and ensures that the document's scroll behavior is only modified when necessary.
+ *
+ * @returns A controller function created by `makeGenericController`.
```

Comment:
Providing `@returns` is no longer needed. The TS return type should give us this info.